expected `i8`, found `u8`

I tried running the example code in my project

```
    tracing_subscriber::registry()
        .with(paranoid_android::layer(
            "com.example.foo",
        ))
        .init();
```

I get the following error:

```
error[E0308]: mismatched types
   --> /Users/justin/.cargo/registry/src/github.com-1ecc6299db9ec823/paranoid-android-0.2.0/src/writer.rs:107:73
    |
107 |                 unsafe { __android_log_buf_write(buffer, priority, tag, message) };
    |                                                                         ^^^^^^^ expected `i8`, found `u8`
    |
    = note: expected raw pointer `*const i8`
               found raw pointer `*const u8`

error[E0308]: mismatched types
   --> /Users/justin/.cargo/registry/src/github.com-1ecc6299db9ec823/paranoid-android-0.2.0/src/writer.rs:209:9
    |
204 |       fn as_ptr(&mut self) -> Option<*const u8> {
    |                               ----------------- expected `Option<*const u8>` because of return type
...
209 | /         CStr::from_bytes_with_nul(self.buf.as_ref())
210 | |             .ok()
211 | |             .map(CStr::as_ptr)
    | |______________________________^ expected `u8`, found `i8`
    |
    = note: expected enum `Option<*const u8>`
               found enum `Option<*const i8>`
```

--- a/src/writer.rs
+++ b/src/writer.rs
@@ -2,6 +2,7 @@ use core::slice;
 use std::{
     ffi::{CStr, CString},
     io::{self, Write},
+    os::raw::c_char,
 };
 
 use lazy_static::lazy_static;
@@ -201,7 +202,7 @@ impl PooledCString {
         self.buf.extend_from_slice(data);
     }
 
-    fn as_ptr(&mut self) -> Option<*const u8> {
+    fn as_ptr(&mut self) -> Option<*const c_char> {
         if self.buf.last().copied() != Some(0) {
             self.buf.push(0);
         }

